### ISSUE TITLE
Crashed

### ISSUE THREAD

**Issue by Curculigo** [SpartanJ/ecode#292]
I setup a C++ project to build with CMake and Ninja. Also tell CMake to generate compile commands. After completed the build to have my compile_commands.json generated, I clicked into main.cpp and it crashed.

dr.mingw report:

```
ecode.exe caused an Access Violation at location 0000000000000000 DEP violation at location 0000000000000000.

AddrPC           Params
0000000000000000 000000000000003F 0000006E3F6ED378 0000006E4870F160
00007FFEA45E4E57 FFFFD00000000103 00007FF7F949E000 0000006E3F6ED1F8  eepp.dll!EE::UI::CSS::StyleSheetPropertyAnimation::tweenProperty+0x637
00007FFEA45E8C81 0000006E414D7960 0000006E48E760A0 0000006E414D7960  eepp.dll!EE::UI::CSS::StyleSheetPropertyAnimation::onUpdate+0x211
00007FFEA45EA3C9 0002625A00000000 000F860400001000 0011FDFF00000001  eepp.dll!EE::UI::CSS::StyleSheetPropertyAnimation::update+0xd9
00007FFEA4353964 0000000000000001 000000000000000C 0000000000000000  eepp.dll!EE::Scene::ActionManager::update+0x234
00007FFEA45584BA 00007FFEA44E56D0 00007FFEA437F04B 0000000000000000  eepp.dll!EE::Scene::SceneNode::update+0x2a
00007FFEA477B52C 000000000000000F 00007FFEA4828D4F 000000000000000F  eepp.dll!EE::UI::UISceneNode::update+0x7c
00007FFEA4553833 0000000000004500 00007FFEA4828E60 0000006E4146FBB0  eepp.dll!EE::Scene::SceneManager::update+0x43
00007FF7FA0AF3F2 0000006E4146FBB0 0000006E3F6ED6B0 0000006E3F6ED630  ecode.exe!0x5f3f2
00007FFEA48294B3 0000006E3F6ED610 0000006E47BC9370 0000006E414700E4  eepp.dll!EE::Window::Window::runMainLoop+0xd3
00007FF7FA0C3E25 0000000000000000 0000006E3F6EEFF8 0000006E3F6EDAB0  ecode.exe!0x73e25
00007FF7FA480427 0000000000000001 0000006E414572F0 0000000000000000  ecode.exe!0x430427
00007FF7FA0512EE 0000000000000000 0000000000000000 0000000000000000  ecode.exe!0x12ee
00007FF7FA0513E6 0000000000000000 0000000000000000 0000000000000000  ecode.exe!0x13e6
00007FFECFA913D2 00007FFECFA913B0 0000000000000000 0000000000000000  KERNEL32.DLL!BaseThreadInitThunk+0x22
00007FFED03F03C4 0000000000000000 0000000000000000 0000000000000000  ntdll.dll!RtlUserThreadStart+0x34
```

**Comment by SpartanJ** [comment]
Thanks for the report. Can you reproduce it intentionally if you try? This one looks odd, it crashed with a CSS animation, I've never seen this.

> I setup a C++ project to build with CMake and Ninja.

You configured the project build settings in ecode or you just did a setup from other application? Anyway given that the error is with an animation it should matter.

Thanks

**Comment by Curculigo** [comment]
I use the built-in build and run feature to call cmake and ninja (multi-steps). I actually use only the Build button since I don't want F5 = Build and Run. I don't want to run the program because I only need it to run cmake (to generate compile_commands.json). I found that if I switch between the build and run tabs and the files tab too quickly, the problem triggers.

**Comment by SpartanJ** [comment]
I wasn't able to reproduce it yet, I'll keep trying. If possible could you record a video crashing ecode? Thanks

**Comment by Curculigo** [comment]
I have encountered this bug again today.

**Comment by Curculigo** [comment]
I have a bunch of `WARNING: Property "is" is not defined!` and `WARNING: applyProperty: Property "is" not defined!` in `ecode.log`. They are not particularly useful, at least for me.

**Comment by Curculigo** [comment]
Can you run a CSS linter on your CSS files in `assets`? It's either that they are faulty or the parser/renderer of eepp is. I hope it's the former.

**Comment by SpartanJ** [comment]
Don't worry about that. It's just a [typo on a property](https://github.com/SpartanJ/eepp/blob/develop/src/tools/ecode/uibuildsettings.cpp#L373) that it's not being used. I fixed a couple of crashes on the Build Settings panel since 0.6.0, this crash is probably related and already fixed but I can't confirm.

**Comment by SpartanJ** [comment]
I finally was able to reproduce it. Thankfully Haiku memory allocator is quite different and reproduced the bug quite easily (it's an invalid memory access). But sadly this bug did not get into the 0.6.1, though since it's quite a bad bug I'll re-upload a patched 0.6.1 into the release packages (so if you already downloaded it, please download it again).
